meetings/summit-nov-2021.md

```diff
@@ -0,0 +1,254 @@
+  * Igor, self-extracting archive
+  * Each platform has its own way of
+* Issues around adding signatures (macOS) 
+* Obfuscation, how important is that?
+  * Is one of the reasons people would want single 
+    binary verus just using script. 
+* Signature issues
+  * Executables have their own format on each platform
+  * They have sections, one more section added to end
+    with signature. 
+  * Cannot just blindly buffer contact, need to make
+    their payload a proper section of the binary
+  * Without signature os does not care if you have
+    added to the binary, signature tools may strip
+    out a section if it is a simple concat.
+  * Packer would need to
+    * Strip signature from Node.js
+    * Add the new section in manner that complies with 
+      OS exe structure
+    * Use signing tools to add new signature
+  * Flow that might makes sense is
+    * On startup Node.js checks for section in
+      exe
+    * Load options from known offset in section
+    * Load JS file from known offset in section
+
+   * Michael Z, be able to run single file in
+     binary format. Then second feature is to 
+     bundle single file.
+     * Joyee can you not already do that with
+       bundlers like Webpack
+     * Michael Z, Ideal if Node.js was its own
+       bundler.
+   * Joyee makes sense for Node.js to be able to patch 
+     itself, just the simple case of patching itself
+     and staring binary js file.
+   * Richard, Issue/PR already to provide framework to
+     allow a virutal file system to be provide
+   * Joyee, bundlers can generate single file
+   * Michael Z, there are limitations, cannot support
+     native modules, or include additional files
+   * Michael, need to define more than just block of
+     JS
+   * Joyee, don't want to contrain what we put in
+     section
+   * Structure needs to have a least 3 sections
+     * options (ex allow --js-flags)  
+     * arguments for Node.js (-e allows js to be specified)
+     * data section to be passed to js file
+   * Command line args, additional arg to add options to
+     Node.js arguments (--js-flags). Force to be the first
+     argument.
+   * Still would not support native addons
+     * pkg ended up placing native modules as dlls shipped
+       along with binary
+     * boxednode includes native addons, we should ask Anna
+       how you do that.
+     * Richard believes boxednode handles native addons by
+       including them in like core modules
+       
+* Approaches
+  * Compile with node, into exe (boxnode)
+  * Bundled onto exising Node.js exe (pkg approach)
+
+* Richard used to be `third party main`, I think we took that
+  out, how do you start something else. Good prior art to consider.
+* Joyee we could add into list of conditions checked in startup
+  for startup before others. `third_party_main.js` was implemented because it was easy (just checking for the existence of it). The bundling solution would be more sophisticated.
```

Comment:
```suggestion
  before others. `third_party_main.js` was implemented because it was easy (just checking for the existence of it). The bundling solution would be more sophisticated.
```